src/components/DotIndicator/DotIndicator.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+
+interface Indicator {
+  ref: number;
+}
+
+interface DotIndicatorProps {
+  current: number;
+  length: number;
+}
+
+const MAX_VISIBLE_INDICATORS = 4;
+
+const DotIndicator = ({ current, length }: DotIndicatorProps) => {
+  const [indicators, setIndicators] = useState<Indicator[]>([]);
+  const [min, setMin] = useState<number>(0);
+  const [max, setMax] = useState<number>(MAX_VISIBLE_INDICATORS - 1);
+
+  useEffect(() => {
+    const newIndicators = [];
+    for (let i = indicators.length || 0; i < length; i++) {
+      newIndicators.push({ ref: i });
+    }
+    setIndicators(newIndicators);
+  }, [length]);
+
+  useEffect(() => {
+    if (current < min) {
+      setMin(current);
+      setMax(current + MAX_VISIBLE_INDICATORS - 1);
+      if (max > length) {
+        setMax(length);
+      }
+    }
+    if (current > max) {
+      setMax(current);
+      setMin(current - MAX_VISIBLE_INDICATORS + 1);
+      if (min < 0) {
+        setMin(0);
+      }
+    }
+  }, [current, length]);
+
+  const getIndicatorClass = (ref: number): string => {
+    if (ref === current) {
+      return "active";
+    }
+    if (ref >= min && ref <= max) {
+      return "std";
+    }
+    if (ref === min - 1 || ref === max + 1) {
+      return "small";
+    }
+    return "hidden";
+  };
+
+  return (
+    <IndicatorsWrapper>
+      {indicators.map((indicator) => (
+        <div
+          key={indicator.ref}
+          className={getIndicatorClass(indicator.ref)}
+        />
+      ))}
+    </IndicatorsWrapper>
+  );
+};
+
+export default DotIndicator;
+
+const IndicatorsWrapper = styled.div<any>`
```

Comment:
[ ASK ] 현재 전달되는 Prop이 없어 타입은 없어도 될것 같은데 추가하신 이유가 있으신지요!
```suggestion
const IndicatorsWrapper = styled.div`
```

Reply:
그러게요!
만들때 <any> 를 넣고 시작했는데 마무리하면서 안지운것 같네요